course6/toyapp/displaylib/surface.h

```diff
@@ -14,9 +15,10 @@ class Surface : private SDLObject
 	Surface(int w, int h);
 	~Surface();
 
-	void Clear(std::uint8_t r, std::uint8_t g, std::uint8_t b);
-	void SetPixel(int x, int y, uint8_t r, uint8_t g, uint8_t b);
-
+	void Clear(Rgb color);
+	void SetPixel(Pixel position, Rgb color);
+	void DrawVertex(Vector3D vertex, Rgb color);
+	void DrawCube(Cube& cube, Rgb color);
```

Comment:
Do you think is a good design? Should Surface know about Cube or 3D Vertex?
I think it should be Cube that knows about the Surface. This way you can expand your catalogue of shapes, without having to teach Surface about every one of them.

Consider moving the methods to they respective objects:

```
Vector3D::Draw(Surface& surf);
Cube::Draw(Surface& surf);
```

Reply:
You're completely right.